src/status_im/common/controlled_input/utils.cljs

```
@@ -23,32 +24,65 @@
   [state error?]
   (assoc state :error? error?))
 
-(defn- upper-limit
+(defn upper-limit
   [state]
   (:upper-limit state))
 
+(defn lower-limit
+  [state]
+  (:lower-limit state))
+
 (defn upper-limit-exceeded?
   [state]
   (and
    (upper-limit state)
    (> (numeric-value state) (upper-limit state))))
 
+(defn lower-limit-exceeded?
+  [state]
+  (and
+   (lower-limit state)
+   (< (numeric-value state) (lower-limit state))))
+
 (defn- recheck-errorness
```

do we need to return the error type? i.e so we can distinguish what message to show?

---

So far we have only one type of error - when the value is out of range, without any specifics. So I think we can add differentiating later if needed.

---

Actually thinking about this, should we just block the input from accepting values that go below 0? I.e the decrease button gets disabled ?